### PR TITLE
DOC: Docstring and examples for Rotation.concatenate

### LINKED ISSUE #19774: DOC: Detail what "concatenate" means in the context of `spatial.transform.Rotation.concatenate`
### Issue with current documentation:

In [the documentation](https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.transform.Rotation.concatenate.html) for `scipy.transform.Rotation.concatenate`, the description `Concatenate a sequence of Rotation objects.` does not actually explain what "concatenate" or "concatenated" means in this context, making it unclear what the functionality does.

### Idea or request for content:

Define the term and its applications in the documentation. An inexperienced user may assume that concatenation == composition; is this true?

Even better (and in addition), include an example(s) in the main [Rotation](https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.transform.Rotation.html) page and or the `concatenate` page.

### Additional context (e.g. screenshots, GIFs)

_No response_

### PR DESCRIPTION
**Pull request by scottshambaugh**
#### Reference issue
Closes https://github.com/scipy/scipy/issues/19774

#### Additional information
Suggested edits to make this more clear welcome!

[skip actions] [skip cirrus]

### CODE REVIEW

**Review by MichaelBonnet — COMMENTED** [review]
this definitely fulfills the requests in my issue #19774. Thanks!

File: scipy/spatial/transform/_rotation.pyx [hunk 1]
```diff
@@ -2294,7 +2294,10 @@ cdef class Rotation:
     @cython.embedsignature(True)
     @classmethod
     def concatenate(cls, rotations):
-        """Concatenate a sequence of `Rotation` objects.
+        """Concatenate a sequence of `Rotation` objects into a single object.
+
+        This is useful if you want to for example take the mean of a set of
+        rotations, and need to pack them into a single object to do so.
```
Comment (j-bowhay) [inline scipy/spatial/transform/_rotation.pyx:2300]:
```suggestion
        This is useful if you want to, for example, take the mean of a set of
        rotations and need to pack them into a single object to do so.
```

**Review by j-bowhay — CHANGES_REQUESTED** [review]
Thanks @scottshambaugh for tackling this so quickly! Some minor nits

**Review by j-bowhay — APPROVED** [review]
Thanks for the quick work @scottshambaugh

**Pull request merged**